src/main/java/com/exasol/containers/ssh/FileVisitor.java

```diff
@@ -0,0 +1,127 @@
+        final Channel channel = this.ssh.openChannel("exec");
+        ((ChannelExec) channel).setCommand(command);
+        final OutputStream out = channel.getOutputStream();
+        final InputStream in = channel.getInputStream();
+        channel.connect();
+        String result = null;
+        while (true) {
+            sendZeroByte(out);
+            if (checkAck(in) != 'C') {
+                break;
+            }
+            readFilemode(in);
+            final long filesize = readFilesize(in);
+            readFilename(in);
+            sendZeroByte(out);
+            result = consumer.process(in, this.ssh.getCharset(), filesize);
+            if (checkAck(in) != 0) {
+                throw new SshException("ack != 0");
+            }
+        }
+
+        channel.disconnect();
+        return result;
+    }
+
+    static void sendZeroByte(final OutputStream stream) throws IOException {
+        stream.write(ZERO_BUFFER, 0, 1);
+        stream.flush();
+    }
+
+    private String readFilemode(final InputStream stream) throws IOException {
+        final byte[] buf = new byte[5];
+        stream.read(buf, 0, 5);
+        return new String(buf, 0, 5);
+    }
+
+    private long readFilesize(final InputStream in) throws IOException {
+        long filesize = 0L;
+        final byte[] buf = new byte[1];
+        while (true) {
+            if (in.read(buf, 0, 1) < 0) {
+                // error
+                return filesize;
+            }
+            if (buf[0] == ' ') {
+                return filesize;
+            }
+            filesize = ((filesize * 10L) + buf[0]) - '0';
+        }
+    }
+
+    private String readFilename(final InputStream stream) throws IOException {
+        final byte[] buf = new byte[1024];
+        for (int i = 0;; i++) {
+            stream.read(buf, i, 1);
+            if (buf[i] == (byte) 0x0a) {
+                return new String(buf, 0, i);
+            }
+        }
+    }
+
+    static String fileHeader(final Path file) throws IOException {
+        return fileHeader('C', "0644", Files.size(file), file.getFileName().toString());
+    }
+
+    static String fileHeader(final char ack, final String mode, final long filesize, final String filename) {
+        return String.format("%c%s %d %s\n", ack, mode, filesize, filename);
+    }
+
+    static int checkAck(final InputStream stream) throws IOException {
+        final int b = stream.read();
+        // b may be 0 for success,
+        // 1 for error,
+        // 2 for fatal error,
+        // -1
+        if (b == 0) {
+            return b;
+        }
+        if (b == -1) {
+            return b;
+        }
+
+        if ((b == 1) || (b == 2)) {
+            final StringBuffer sb = new StringBuffer();
+            int c;
+            do {
+                c = stream.read();
+                sb.append((char) c);
+            } while (c != '\n');
+            if (b == 1) { // error
+                LOGGER.error(sb.toString());
```

Comment:
Throw an exception in this case?

Reply:
I copied the code from http://www.jcraft.com/jsch/examples/ScpFrom.java.html and wanted to avoid to change too much.